src/main/java/com/aws/greengrass/mqttclient/AwsIotMqttClient.java

```diff
@@ -51,7 +57,7 @@
  * Wrapper for a single AWS IoT MQTT client connection.
  * Do not use except through {@link MqttClient}.
  */
-class AwsIotMqttClient implements Closeable {
+class AwsIotMqttClient implements IndividualMqttClient {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This class references 46 other classes. By comparison, 98% of the classes in the CodeGuru reference dataset reference fewer. This indicates that this class is highly coupled with other classes. A class that is highly coupled with other classes is difficult to understand and its behavior might change unexpectedly when one of its referenced classes is updated. High coupling could also increase the integration test complexity, maintenance cost and technical debt. We recommend that you simplify this class or break it into multiple classes.